lib/rouge/lexers/csharp.rb

```diff
@@ -47,19 +47,38 @@ class CSharp < RegexLexer
         rule %r(/[*].*?[*]/)m, Comment::Multiline
       end
 
+      state :nest do
+        rule /{/, Punctuation, :nest
+        rule /}/, Punctuation, :pop!
+        mixin :root
+      end
+
+      state :splice_string do
+        rule /\\./, Str
+        rule /{/, Punctuation, :nest
+        rule /"|\n/, Str, :pop!
+        rule /./, Str
+      end
+
+      state :splice_literal do
+        rule /""/, Str
```

Comment:
That would be inconsistent with non-spliced literals, as well as with escapes in other strings (in C#). How about leaving this as-is and then creating a separate PR for adding Str::Escape everywhere? (I can't promise to do that this weekend but I'll aim for next).

By the way, I couldn't get escapes to show up differently in the test interface. I didn't try too hard, though.